static/app/views/app/index.tsx

```diff
@@ -35,201 +31,151 @@ const GlobalNotifications = HookOrDefault({
+  const api = useApi();
+  const {organization} = useLegacyStore(OrganizationStore);
 
-    ConfigStore.get('messages').forEach(msg => {
-      AlertActions.addAlert({
-        message: msg.message,
-        type: msg.level,
-        neverExpire: true,
-      });
-    });
+  // Command palette global-shortcut
+  useHotkeys('command+shift+p, command+k, ctrl+shift+p, ctrl+k', e => {
+    openCommandPalette();
+    e.preventDefault();
+  });
+
+  // Theme toggle global shortcut
+  useHotkeys(
+    'command+shift+l, ctrl+shift+l',
+    e => {
+      ConfigStore.set('theme', config.theme === 'light' ? 'dark' : 'light');
+      e.preventDefault();
+    },
+    [config.theme]
+  );
 
-    if (DEPLOY_PREVIEW_CONFIG) {
-      displayDeployPreviewAlert();
-    } else if (EXPERIMENTAL_SPA) {
-      displayExperimentalSpaAlert();
+  /**
+   * Loads the users organization list into the OrganizationsStore
+   */
+  async function loadOrganizations() {
+    try {
+      const data = await api.requestPromise('/organizations/', {query: {member: '1'}});
+      OrganizationsStore.load(data);
+    } catch {
+      // TODO: do something?
     }
+  }
 
-    initApiClientErrorHandling();
+  /**
+   * Creates Alerts for any internal health problems
+   */
+  async function checkInternalHealth() {
+    let data: any = null;
 
-    const user = ConfigStore.get('user');
-    if (user) {
-      HookStore.get('analytics:init-user').map(cb => cb(user));
+    try {
+      data = await api.requestPromise('/internal/health/');
+    } catch {
+      // TODO: do something?
     }
 
-    fetchGuides();
-  }
+    data?.problems?.forEach?.(problem => {
+      const {id, message, url} = problem;
+      const type = problem.severity === 'critical' ? 'error' : 'warning';
 
-  componentDidUpdate(prevProps) {
-    const {config} = this.props;
-    if (!isEqual(config, prevProps.config)) {
-      this.handleConfigStoreChange(config);
-    }
+      AlertActions.addAlert({id, message, type, url});
+    });
   }
 
-  componentWillUnmount() {
-    OrganizationsStore.load([]);
-    this.unlistener?.();
-  }
+  useEffect(() => {
+    loadOrganizations();
+    checkInternalHealth();
 
-  mainContainerRef = createRef<HTMLDivElement>();
-  unlistener = OrganizationStore.listen(
-    state => this.setState({organization: state.organization}),
-    undefined
-  );
+    // Show system-level alerts
+    config.messages.forEach(msg =>
+      AlertActions.addAlert({message: msg.message, type: msg.level, neverExpire: true})
+    );
 
-  handleConfigStoreChange(config) {
-    const newState = {} as State;
-    if (config.needsUpgrade !== undefined) {
-      newState.needsUpgrade = config.needsUpgrade;
+    // The app is running in deploy preview mode
+    if (DEPLOY_PREVIEW_CONFIG) {
+      displayDeployPreviewAlert();
     }
 
-    if (config.user !== undefined) {
-      newState.user = config.user;
+    // The app is running in local SPA mode
+    if (!DEPLOY_PREVIEW_CONFIG && EXPERIMENTAL_SPA) {
```

Comment:
I broke up this conditional from using `else if` to make it marginally more readable.